src/properties.jl

```diff
@@ -0,0 +1,668 @@
+    class_getproperty(superclass(FileAccessProperties), p, name)
+end
+function class_setproperty!(::Type{FileAccessProperties}, p::Properties, name::Symbol, val)
+    name === :alignment     ? set_alignment!(p, val) :
+    name === :driver        ? Drivers.set_driver!(p, val) :
+    name === :fclose_degree ? set_fclose_degree!(p, val) :
+    name === :libver_bounds ? set_libver_bounds!(p, val) :
+    # deprecated
+    name === :fapl_mpio     ? (depwarn("The `fapl_mpio` property is deprecated, use `driver=HDF5.Drivers.MPIO(...)` instead.", :fapl_mpio); p.driver = Drivers.MPIO(val...)) :
+    class_setproperty!(superclass(FileAccessProperties), p, name, val)
+end
+
+
+@propertyclass LinkAccessProperties API.H5P_LINK_ACCESS
+
+@propertyclass GroupAccessProperties API.H5P_GROUP_ACCESS
+superclass(::Type{GroupAccessProperties}) = LinkAccessProperties
+
+@propertyclass DatatypeAccessProperties API.H5P_DATATYPE_ACCESS
+superclass(::Type{DatatypeAccessProperties}) = LinkAccessProperties
+
+@propertyclass DatasetAccessProperties API.H5P_DATASET_ACCESS
+superclass(::Type{DatasetAccessProperties}) = LinkAccessProperties
+
+@propertyclass AttributeAccessProperties API.H5P_ATTRIBUTE_ACCESS
+superclass(::Type{AttributeAccessProperties}) = LinkAccessProperties
+
+"""
+    DatasetTransferProperties(;kws...)
+
+Properties used when transferring data to/from datasets
+
+- `dxpl_mpio`: MPI transfer mode:
+   - `:independent`: use independent I/O access (default),
+   - `:collective`: use collective I/O access.
+"""
+@propertyclass DatasetTransferProperties API.H5P_DATASET_XFER
+
+@enum_property(dxpl_mpio,
+               :independent => API.H5FD_MPIO_INDEPENDENT,
+               :collective  => API.H5FD_MPIO_COLLECTIVE)
+
+class_propertynames(::Type{DatasetTransferProperties}) = (
+    :dxpl_mpio,
+    )
+function class_getproperty(::Type{DatasetTransferProperties}, p::Properties, name::Symbol)
+    name === :dxpl_mpio  ? get_dxpl_mpio(p) :
+    class_getproperty(superclass(DatasetTransferProperties), p, name)
+end
+function class_setproperty!(::Type{DatasetTransferProperties}, p::Properties, name::Symbol, val)
+    name === :dxpl_mpio  ? set_dxpl_mpio!(p, val) :
+    class_setproperty!(superclass(DatasetTransferProperties), p, name, val)
+end
+
+@propertyclass FileMountProperties API.H5P_FILE_MOUNT
+@propertyclass ObjectCopyProperties API.H5P_OBJECT_COPY
+
+
+const DEFAULT_PROPERTIES = GenericProperties()
+# These properties are initialized in __init__()
+const ASCII_LINK_PROPERTIES = LinkCreateProperties()
+const UTF8_LINK_PROPERTIES = LinkCreateProperties()
+const ASCII_ATTRIBUTE_PROPERTIES = AttributeCreateProperties()
+const UTF8_ATTRIBUTE_PROPERTIES = AttributeCreateProperties()
+
+_link_properties(::AbstractString) = UTF8_LINK_PROPERTIES
+_attr_properties(::AbstractString) = UTF8_ATTRIBUTE_PROPERTIES
```

Comment:
Can we just remove these and pass in the global constants directly.  To me the way these are used don't offer any advantages and is less clear than using the global constants. Thoughts on this?

Reply:
Yes, that would probably make sense (I'm not 100% sure what I was thinking here)